Updated `Release_notes.md` for 2025 Q1 release

dmccoystephenson opened the pull request:
# PR Details
## Description
A new version of jpo-ode will be released this month, but the release notes need to be updated. The updated release notes now include all changes made by CDOT for this quarter, as well as planned changes from USDOT.

## Related Issue
No related GitHub issue.

## Motivation and Context
This change is required to ensure that the release notes accurately reflect all recent updates and improvements made to jpo-ode. By incorporating changes from both CDOT and planned updates from USDOT, the release notes provide a comprehensive overview of the current state of the software.

## How Has This Been Tested?
Documentation update, no testing necessary.

## Types of changes
- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)
- [x] Documentation

## Checklist:
- [ ] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[ODE Contributing Guide](https://github.com/usdot-jpo-ode/jpo-ode/blob/bugfix/Pull_request_template/docs/contributing_guide.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

drewjj commented on docs/Release_notes.md:
```diff
@@ -1,13 +1,57 @@
-This is a placeholder for the upcoming 4.0.0 release. The release notes will be updated with the details of the release once it is available.
 
-The major version was incremented due to breaking changes in the 2024 revision of J2735.
+The jpo-ode 4.0.0 release brings a variety of updates aimed at enhancing functionality, maintainability, and the overall development experience. Key highlights include the introduction of a Generated TMC TIM Topic, Spring Kafka migrations for improved integration, and updates to TIM schemas and JSON annotations for more accurate ODE processing. Significant strides have been made in optimizing con ...
```
```suggestion
The jpo-ode 4.0.0 release brings a variety of updates aimed at enhancing functionality, maintainability, and the overall development experience. Key highlights include the introduction of a generated TMC TIM Topic, the migration from custom a Kafka client to Spring Kafka for improved integration, and updates to TIM schemas and JSON annotations for more J2735-accurate ODE processing. Significant strides have been made in optimizing configurations, with enhancements like the extraction of Kafka properties, environment variables, and Kafka topics into more streamlined Spring Configuration properties. Additionally, schema updates have been made to increment the version for output messages and ensure compatibility with J2735 2024. Developer-focused changes include revisions to the README, Makefile, and devcontainer to support smoother onboarding and ease of use. Other important updates encompass new entries in the submodule compatibility guide, version upgrades, and bug fixes related to Kafka configurations and message handling.
```
Just some small recommendations to capitalization and making certain points clearer to someone without knowledge of our team lingo.

drewjj commented on docs/Release_notes.md:
```diff
@@ -1,13 +1,57 @@
+The jpo-ode 4.0.0 release brings a variety of updates aimed at enhancing functionality, maintainability, and the overall development experience. Key highlights include the introduction of a Generated TMC TIM Topic, Spring Kafka migrations for improved integration, and updates to TIM schemas and JSON annotations for more accurate ODE processing. Significant strides have been made in optimizing con ...
 
+Enhancements in this release:
+- [CDOT PR 100](https://github.com/CDOT-CV/jpo-ode/pull/100): Add Generated TMC TIM Topic
```
I like the inclusion of links to the PRs! Definitely useful when this is going to be located on a USDOT repository where these will not be visible from the pull requests tab.

mcook42 reviewed (commented):
Question(blocking): I see we've updated `docs/Release_notes.md` but we also have a file `ReleaseNotes.md` at the top level of this repo. Which one should we be maintaining, or do we need to maintain both files?

dmccoystephenson commented:
> Question(blocking): I see we've updated `docs/Release_notes.md` but we also have a file `ReleaseNotes.md` at the top level of this repo. Which one should we be maintaining, or do we need to maintain both files?

Good point! We only maintain docs/Release_notes.md, so I’ve moved the contents of ReleaseNotes.md to this file and deleted the outdated one.

mcook42 commented on docs/Release_notes.md:
```diff
@@ -275,5 +326,193 @@ The New Ode version 1.0.8 includes below updates:
+
+### Sprint 15
+- ODE-381 Leidos Security Library Integration implementing 1609.2
+  -- As part of this new feature, the build process was updated in the [ODE README file](README.md) with additional steps for obtaining the source code and building the application. Please review the [Getting Started](https://github.com/usdot-jpo-ode/jpo-ode#getting-started) section for details.
```
suggestion(non-blocking): Update the link to the readme
```suggestion
  -- As part of this new feature, the build process was updated in the [ODE README file](../README.md)) with additional steps for obtaining the source code and building the application. Please review the [Getting Started](https://github.com/usdot-jpo-ode/jpo-ode#getting-started) section for details.
```

mcook42 commented on docs/Release_notes.md:
```diff
@@ -1,13 +1,62 @@
-This is a placeholder for the upcoming 4.0.0 release. The release notes will be updated with the details of the release once it is available.
 
-The major version was incremented due to breaking changes in the 2024 revision of J2735.
+The jpo-ode 4.0.0 release brings a variety of updates aimed at enhancing functionality, maintainability, and the overall development experience. Key highlights include the introduction of a generated TMC TIM Topic, the migration from a custom Kafka client to Spring Kafka for improved Spring integration, and updates to TIM schemas and JSON annotations for more J2735-accurate ODE processing. Signif ...
```
suggestion(non-blocking): 
- Use a formatted list for the key highlights to improve readability
- Wrap long strings of text to improve plaintext reviewing of the file. Markdown renderers auto-wrap text, but if readers don't use one, then they have to scroll to the right to read the text

```suggestion
The jpo-ode 4.0.0 release brings a variety of updates aimed at enhancing functionality, maintainability, and overall development experience.

Key highlights include:
- The introduction of a generated TMC TIM Topic
- The migration from a custom Kafka client to Spring Kafka for improved Spring integration
- Updates to TIM schemas and JSON annotations for more J2735-accurate ODE processing

Significant strides have been made in optimizing configurations through enhancements like the extraction of Kafka properties,
environment variables, and Kafka topics into more streamlined Spring Configuration properties. Additionally, schema
updates have been made to increment the version for output messages and ensure compatibility with J2735 2024.
Developer-focused changes include revisions to the README, Makefile, and devcontainer to support smoother onboarding and
ease of use. Other vital updates encompass new entries in the submodule compatibility guide, version upgrades, and
bug fixes related to Kafka configurations and message handling.
```

mcook42 reviewed (approved):
A few non-blocking suggestions. Otherwise, looks great
